indico/modules/events/editing/notifications.py

```diff
@@ -0,0 +1,88 @@
+    """
+    submitter = revision.submitter
+    editor = revision.editable.editor
+    tpl = get_template_module('events/editing/emails/submitter_comment_notification.txt',
+                              submitter_name=submitter.first_name, timeline_url=_get_timeline_url(revision),
+                              recipient_name=editor.first_name)
+    return make_email(editor.email, template=tpl)
+
+
+@email_sender
+def notify_editor_comment(revision, admin=False):
+    """
+    Notifies the submitter about comments made by an editor.
+    """
+    submitter = revision.submitter
+    editor = session.user if admin else revision.editable.editor
+    tpl = get_template_module('events/editing/emails/editor_comment_notification.txt',
+                              editor_name=editor.first_name, timeline_url=_get_timeline_url(revision),
+                              recipient_name=submitter.first_name)
+    return make_email(submitter.email, template=tpl)
+
+
+@email_sender
+def notify_editor_judgement(revision):
+    """
+    Notifies the submitter about a judgement made by an editor.
+    """
+    submitter = revision.submitter
+    editor = revision.editable.editor
+    tpl = get_template_module('events/editing/emails/editor_judgement_notification.txt',
+                              editor_name=editor.first_name, timeline_url=_get_timeline_url(revision),
+                              recipient_name=submitter.first_name)
+    return make_email(submitter.email, template=tpl)
+
+
+@email_sender
+def notify_submitter_upload(revision):
+    """
+    Notifies the editor about the submitter uploading a new revision.
+    """
+    submitter = revision.submitter
+    editor = revision.editable.editor
+    tpl = get_template_module('events/editing/emails/submitter_upload_notification.txt',
+                              submitter_name=submitter.first_name, timeline_url=_get_timeline_url(revision),
+                              recipient_name=editor.first_name)
+    return make_email(editor.email, template=tpl)
+
+
+@email_sender
+def notify_submitter_confirmation(revision):
+    """
+    Notifies the editor(s) about submitter accepting/rejecting revision changes.
+    """
+    submitter = revision.submitter
+    editable = revision.editable
+    current_editor = editable.editor
+    prev_revision_editor = editable.revisions[-2].editor
+    recipients = {current_editor, prev_revision_editor}
+    template_path = 'events/editing/emails/submitter_confirmation_notification.txt'
+    return [make_email(recipient.email,
+                       template=get_template_module(template_path,
+                                                    submitter_name=submitter.first_name,
+                                                    timeline_url=_get_timeline_url(revision),
+                                                    recipient_name=recipient.first_name)
+                       ) for recipient in recipients]
```

Comment:
This one actually becomes more readable when either not using the decorator or using it with `yield`:

```python
    for recipient in recipients:
        tpl = get_template_module('events/editing/emails/submitter_confirmation_notification.txt',
                                  submitter_name=submitter.first_name,
                                  timeline_url=_get_timeline_url(revision),
                                  recipient_name=recipient.first_name)
        yield make_email(recipient.email, template=tpl)  
```